tests/virt_autotest/sriov_pci_passthrough.pm

```diff
@@ -50,88 +45,97 @@ sub run_test {
   $self->{test_results}->{"host"}->{"Error: fail to enable the SR-IOV PF @host_pfs!"}->{status} = 'FAILED' if ! @host_vfs;
   record_info("VFs enabled", "@host_vfs");
 
-  #generate ssh key on host
-  script_run "rm ~/.ssh/id_rsa*; ssh-keygen -t rsa -f ~/.ssh/id_rsa -P ''";
-
-  foreach my $guest (keys %xen::guests) {
-  my $guest = "sles-15-sp2-64-fv-def-net";
+  #foreach my $guest (keys %xen::guests) {
+    my $guest = "sles-15-sp2-64-fv-def-net";
 
     prepare_guest($guest);
     
-    #detach one of the vfs from host
-    my %host_vf;
-    $host_vf{bdf} = $host_vfs[int(rand(7))];  #pick up randomly
-    $host_vf{id} = get_device_id($host_vf{bdf});
-    record_info("Detach VF from host", "$host_vf{id}");
-    assert_script_run "virsh nodedev-detach $host_vf{id}";
-
-    #plugin vf to vm via virsh
-    record_info("Pluging vf to vm via libvirt", "guest=$guest \nvf=$host_vf{id}");
-    my %vf_vm = plugin_device_kvm($guest, $host_vf{id});
-    record_info("VF plugg'ed in", "bdf=$vf_vm{bdf}, mac=$vf_vm{mac}");
+    #detach the first vf from host
+    my %host_vf_1;
+    $host_vf_1{bdf} = $host_vfs[int(rand($#host_vfs + 1))];  #pick up randomly
+    $host_vf_1{id} = detach_vf_from_host($host_vf_1{bdf});
+
+    #detach the second vf from host
+    my %host_vf_2;
+    $host_vf_2{bdf} = $host_vfs[int(rand($#host_vfs + 1))];
+    while ($host_vf_2{bdf} eq $host_vf_1{bdf}){
+      $host_vf_2{bdf} = $host_vfs[int(rand($#host_vfs + 1))];
+    }
+    $host_vf_2{id} = detach_vf_from_host($host_vf_2{bdf});
+
+    #plugin vfs to vm via virsh
+    my %vf_vm_1 = plugin_device($guest, $host_vf_1{id});
+    my %vf_vm_2 = plugin_device($guest, $host_vf_2{id});
   
     #check the networking of the plugged interface
     my $gateway = script_output "ip r s | grep 'default via' | cut -d ' ' -f3";
     set_var("SRIOV_PCI_PASSSHTROUGH", 1);   #to differenciate virtual network tests
-    test_network_interface($guest, gate => $gateway, mac => $vf_vm{mac}, net => 'br123');   #use br123 as ssh connection
+    test_network_interface($guest, gate => $gateway, mac => $vf_vm_1{mac}, net => 'br123');   #use br123 as ssh connection
+    test_network_interface($guest, gate => $gateway, mac => $vf_vm_2{mac}, net => 'br123');
   
-    #reboot
+    #reboot the guest
     record_info("VM rebooting", "$guest");
-    script_run "ssh root\@$guest 'reboot'";
-    my $time = 0;
-    while($time < 120 ) {
-      last if ! script_run "ssh root\@$guest 'echo Hello!'";
-      $time += 10;
-      sleep 10;
-    }
+    assert_script_run "ssh root\@$guest 'reboot'";
+    script_retry("nmap $_ -PN -p ssh | grep open", delay => 10, retry => 18,  die => 1);
   
```

Comment:
Can it be more explicit for $_?

Reply:
Already been fixed in my yesterday's last revision. you may view an outdated one.